File: Challenges/450_DSA_Cracker/Array/024_Find_Longest_Consecutive_Subsequence.java
//{ Driver Code Starts
//Initial Template for Java

import java.math.*;
import java.util.*;
import java.io.*;

class Driverclass
{
    // Driver Code
    static class FastReader{ 
        BufferedReader br; 
        StringTokenizer st; 
  
        public FastReader(){ 
            br = new BufferedReader(new InputStreamReader(System.in)); 
        } 
  
        String next(){ 
            while (st == null || !st.hasMoreElements()){ 
                try{ st = new StringTokenizer(br.readLine()); } catch (IOException  e){ e.printStackTrace(); } 
            } 
            return st.nextToken(); 
        } 
  
        String nextLine(){ 
            String str = ""; 
            try{ str = br.readLine(); } catch (IOException e) { e.printStackTrace(); } 
            return str; 
        } 

        Integer nextInt(){
            return Integer.parseInt(next());
        }
    }
    
	public static void main(String args[])
	{
		FastReader sc = new FastReader();
		PrintWriter out = new PrintWriter(System.out);
		int t = sc.nextInt();
		
		while(t>0)
		{
			int n = sc.nextInt();
			int a[] = new int[n];
			
			for(int i=0; i<n; i++)
				a[i] = sc.nextInt();
		    out.println(new Solution().findLongestConseqSubseq(a, n));
		    t--;
		}
		out.close();
	}
}
// } Driver Code Ends


//User function Template for Java

class Solution
{   
    // arr[] : the input array
    // N : size of the array arr[]
    static int limit = 100001;
    
	static int findLongestConseqSubseq(int arr[], int N)
	{
	   int arr2[] = new int[limit];
	   for(int i=0; i<N; i++) {
	       arr2[arr[i]] = 1;
	   }
	   
	   int l = 0, ml = 0;
	   for(int i=0; i<limit; i++) {
	       if(arr2[i] == 0) {
	           l = 0;
	       } else {
	           l += 1;
	       }
	       ml = Math.max(ml, l);
	   }
	   
	   return ml;
	}
}

File: Challenges/450_DSA_Cracker/Array/024_Find_Longest_Consecutive_Subsequence.py
 #User function Template for python3
 
class Solution:
    
    # arr[] : the input array
    # N : size of the array arr[]
    
    def findLongestConseqSubseq(self,arr, N):
        s = set()
        for i in arr : 
            s.add(i)
        
        ml = 0
        for i in s : 
            if i-1 in s : 
                continue
            else :
                c = 0
                ele = i
                while ele in s : 
                    c += 1
                    ele += 1
                ml = max(c, ml)
        return ml


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__ == '__main__':
    t = int(input())
    for tt in range(t):
        n=int(input())
        a = list(map(int, input().strip().split()))
        print(Solution().findLongestConseqSubseq(a,n))
# } Driver Code Ends
